src/dial_number.c

```diff
@@ -18,7 +18,7 @@ int clean_number(char* str)
 	 * In other cases trust the user input
 	 */
 	while (str[i]) {
-		if (isalpha(str[i]) != 0)
+		if (isalpha((int) str[i]) != 0)
```

Comment:
not sure why this is needed ...

passing a char to an int function should be legal, as long as sizeof(char) is less than sizeof(int).

What kind of platform is Zephyr, is it linux based ?

Reply:
It is RTOS based.

It is a gcc warning:
```
/home/cspiel/dev/zephyrproject/modules/sip/baresip/lib/src/dial_number.c:21:32: warning: array subscript has type 'char' [-Wchar-subscripts]
   21 |                 if (isalpha(str[i]) != 0)
```

Should I turn off the warning instead?

Reply:
Example valid code:

```
int a;
char b = 'X';

a = b;
```

You might want to check the size of these types on your system.